### PR TITLE
Updated Documentation gh-pages for Invalid exchange in REMREM return SUCCESS.

### PR DESCRIPTION
**Pull request by ZVADHEM**
Updated Documentation for Invalid exchange
in REMREM return SUCCESS when trying to
publish a Eiffel Message

### Description of the Change

Added a new option ce or create exchange to create an exchange if it not available. it is passed as 
command line args for cli (true/false). 
createExchangeIfNotExisting: true or false property is added for both spring and tomcat properties file.

### Sign-off

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.

Signed-off-by:

### CODE REVIEW

File: cliUsage.html [hunk 1]
```diff
@@ -56,6 +56,9 @@ <h2>
 
 -en,--exchange_name &lt;arg&gt;         Exchange name.
 
+-ce,--create_exchange &lt;arg&gt;       If Exchange Doesn't exist we can create exchange by setting
+                                  create_exchange or ce to true by setting as command line args eg: -ce true/false.
```
Comment (SantoshNC68) [inline cliUsage.html]: didnot understand this statement, modify it.

File: cliUsage.html [hunk 2]
```diff
@@ -135,6 +138,9 @@ <h5>If you want to have the message publishing on RabbitMQ with given routing ke
         <h5>If you want to have the message publishing on RabbitMQ with given tag:</h5>
         <pre><code>java -jar publish-cli.jar -f publishMessages.json -en rabbit -mb mb123 -mp eiffelsemantics -tag production</code></pre>
 
+        <h5>If you want to have the message publishing on RabbitMQ by creating an exchange with given create_exchange:</h5>
```
Comment (SantoshNC68) [inline cliUsage.html]: If there are no exchanges, then we can create an exchange and publish the messages to it by using create_exchange property. Example:

File: cliUsage.html [hunk 3]
```diff
@@ -145,6 +151,14 @@ <h5>If you want to have the message publishing on RabbitMQ with given tag:</h5>
   }
 ]</pre>
 
+        <p>Typical output for createExchange when set to false and exchange is not available in RabbitMQ example is:</p>
```
Comment (SantoshNC68) [inline cliUsage.html]: remove is in the statement, and add a ',' after RabbitMQ

File: cliUsage.html [hunk 4]
```diff
@@ -145,6 +151,14 @@ <h5>If you want to have the message publishing on RabbitMQ with given tag:</h5>
...
+<pre>[
+  {
+    "status_code": 404,
+    "result": "ExchangeName Exchange not found, Please check exchange configuration and try again",
```
Comment (SantoshNC68) [inline cliUsage.html]: mention the ExchangeName in <> braces.
Same for the next line.

File: serviceUsage.html [hunk 5]
```diff
@@ -58,12 +58,18 @@ <h4>Running and configuring RemRem-Publish without Tomcat installation</h4>
...
         <p>It is also possible to provide path to the spring properties file:</p>
-        <pre>java -jar publish-service/target/publish-service-x.x.x.war --spring.config.location=path/to/applicattion.propeties</pre>
+        <pre>java -jar publish-service/target/publish-service-x.x.x.war --spring.config.location=path/to/application.properties</pre>
 
         <p>Second option to execute RemRem-Publish is to execute maven commands from the source code root folder, example:</p>
         <pre>mvn spring-boot:run -Dspring-boot.run.arguments=--properties.parameter1=value,--properties.parameter2=value,--properties.parameter3=value,--properties.parameter4=value ......</pre>
         <p>When using maven command, Spring properties can also be changed by editing publish-service/src/main/resources/application.properties file.</p>
 
+                <p>An <a href="https://www.rabbitmq.com/tutorials/tutorial-three-java.html">exchange point</a> must exist before starting the service.
+        if not application will not be up and running.
```
Comment (SantoshNC68) [inline serviceUsage.html]: I capital, this can be added as a continuation to previous statement.

File: cliUsage.html [hunk 6]
```diff
@@ -56,6 +56,9 @@ <h2>
 
 -en,--exchange_name &lt;arg&gt;         Exchange name.
 
+-ce,--create_exchange &lt;arg&gt;       option to denote if we need to create an exchange
```
Comment (xvinosi-github) [inline cliUsage.html]: option to denote exchange creation ex: -ce true or --create_exchange false

File: cliUsage.html [hunk 7]
```diff
@@ -103,13 +106,13 @@ <h2>
         <p>Typical examples of usage Eiffel REMReM Publish CLI are described below.</p>
 
         <h5>Publish on a given host, given exchange, given domain and given user domain suffix:</h5>
-        <pre><code>java -jar publish-cli.jar -f publishMessages.json -en rabbit -mb mb101 -domain eiffel001 -ud fem001 -mp eiffelsemantics</code></pre>
+        <pre><code>java -jar publish-cli.jar -f publishMessages.json -en exchangeName -mb mb101 -domain eiffel001 -ud fem001 -mp eiffelsemantics</code></pre>
```
Comment (xvinosi-github) [inline cliUsage.html]: mb101-eiffel1001

File: cliUsage.html [hunk 8]
```diff
@@ -118,24 +121,27 @@ <h5>For loading protocol jars other than eiffelsemantics:</h5>
...
 
         <h5>If you want to have the message publishing on RabbitMQ with given routing key:</h5>
-        <pre><code>java -jar publish-cli.jar -f publishMessages.json -en rabbit -mb mb123 -mp eiffelsemantics -rk myroutingkey</code></pre>
+        <pre><code>java -jar publish-cli.jar -f publishMessages.json -en exchangeName -mb localhost:5672 -mp eiffelsemantics -rk myroutingkey</code></pre>
 
         <h5>If you want to have the message publishing on RabbitMQ with given tag:</h5>
-        <pre><code>java -jar publish-cli.jar -f publishMessages.json -en rabbit -mb mb123 -mp eiffelsemantics -tag production</code></pre>
+        <pre><code>java -jar publish-cli.jar -f publishMessages.json -en exchangeName -mb localhost:5672 -mp eiffelsemantics -tag production</code></pre>
+
+        <h5>If there are no exchanges, then we can create an exchange and publish the messages to it by using create_exchange property. Example:</h5>
```
Comment (xvinosi-github) [inline cliUsage.html]: To create an exchange passed while publishing the messages use --create_exchange or -ce to true.

File: cliUsage.html [hunk 9]
```diff
@@ -145,6 +151,14 @@ <h5>If you want to have the message publishing on RabbitMQ with given tag:</h5>
   }
 ]</pre>
 
+        <p>Typical output for createExchange when set to false and exchange is not available in RabbitMQ, Example:</p>
```
Comment (xvinosi-github) [inline cliUsage.html]: Output when exchange is not found and --create_exchange is set to false.

File: serviceUsage.html [hunk 10]
```diff
@@ -128,9 +134,13 @@ <h3>Exchange configurations</h3>
 &lt;protocol&gt;.rabbitmq.password:      &lt;password, default for RabbitMQ Server: guest&gt;
 &lt;protocol&gt;.rabbitmq.tls:           &lt;tls version, is optional&gt;
 &lt;protocol&gt;.rabbitmq.exchangeName:  &lt;exchange name, exchange should be already created on RabbitMQ Server&gt;
+&lt;protocol&gt;.rabbitmq.createExchangeIfNotExisting:  &lt;true or false, Exchange will be created when set to true on RabbitMQ Server&gt;
```
Comment (xvinosi-github) [inline serviceUsage.html]: Create Exchange if not present on RabbitMQ.

File: serviceUsage.html [hunk 11]
```diff
@@ -128,9 +134,13 @@ <h3>Exchange configurations</h3>
...
 &lt;protocol&gt;.rabbitmq.domainId:      &lt;domain id, any string&gt;</pre>
 
-        <p>An <a href="https://www.rabbitmq.com/tutorials/tutorial-three-java.html">exchange point</a> must exist before starting the service.</p>
+        <p>An <a href="https://www.rabbitmq.com/tutorials/tutorial-three-java.html">exchange point</a> must exist before starting the service.
```
Comment (xvinosi-github) [inline serviceUsage.html]: Why repeat the content, can be mentioned only once. Remove this from top.
Change as below:
starting the  service if createExchangeIfNotExisting is set to false or application launch is terminated.
Application can create the exchange by adding the below property in config.properties file.

**Pull request merged**